### PR TITLE
refactor PCA tests

### PR DESCRIPTION
**Pull request by kain88-de**
Changes made in this Pull Request:
 - use pytest fixtures
 - improve runtime from 12sec to 1sec on my laptop
 - increase coverage

Using the module level fixture for `pca` was a huge time saver. decreasing the number of atoms to calculate the PCA for also gave it a huge boost. This trim the overall test-time by ~1min or more. Without sacrificing test quality! I'll check if I can get coverage up to 100%

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - ~~[ ] Issue raised/referenced?~~

### CODE REVIEW

**Comment by richardjgowers** [comment]
This is pretty cool.  The only concern with this is that if one test modifies the fixture, it will remain modified in all subsequent tests right?

**Comment by kain88-de** [comment]
yes. But that is OK here. Of course we should take care of using a module fixture. But this is a very limited number of tests where it's easy to check that the fixture isn't changed.

File: testsuite/MDAnalysisTests/analysis/test_pca.py [hunk 1]
```diff
@@ -21,78 +21,99 @@
...
+    assert_equal(pca_space.shape, (u.trajectory.n_frames, 1))
+
+
+def test_transform_mismatch(pca, u):
+    with pytest.raises(ValueError):
+        pca.transform(u, n_components=1)
+
+
+def test_transform_universe():
+    u1 = mda.Universe(waterPSF, waterDCD)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/analysis/test_pca.py]: So for example here, if we're concerned that `u1` gets altered, we could make a module scoped version of the Universe, then make function scoped fixtures which return a copy.  As a toy example, the below code only works with the `copy.copy` call, removing this makes the array in the `this` fixture get incremented too many times

```python
import numpy as np
import copy
import pytest

@pytest.fixture(scope='module')
def this():
    print('making a')
    a = np.zeros(10)
    a[:] = 10
    return a[:]

@pytest.fixture(scope='function')
def that(this):
    my_this = copy.copy(this)
    my_this += 1
    return my_this


def test_that(that):
    assert all(that == 11)

def test_that_2(that):
    assert all(that == 11)

def test_that_3(that):
    assert all(that == 11)


```

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_pca.py]: Couldn't we then make our own fixture decorator that returns a *cheap* copy?

Comment (mnmelo) [inline testsuite/MDAnalysisTests/analysis/test_pca.py]: `copy.copy` is the already the cheap version, right?
But I'm not sure a shallow copy helps us much here: a universe is essentially a collection of references. And those aren't duplicated by the shallow copy. A `copy.deepcopy` is a safer guard against tests modifying the fixture, but likely much slower. If the deepcopying time is close to universe generation from file, then using a fixture is moot.

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_pca.py]: Maybe we can open a new issue for this discussion

**Comment by kain88-de** [comment]
Oh I should mention that we save a lot of time here because we minimize calculating the PCA which involves an expensive eigenvalue decomposition. Not because of the universe fixture. I could write the tests without the module scoped universe fixture and get very similar speeds

**Pull request merged**